python-sqlite-sqlalchemy/project/examples/example_3/app/albums/routes.py

```diff
@@ -0,0 +1,70 @@
+from flask import Blueprint, render_template, redirect, url_for
+from flask_wtf import FlaskForm
+from wtforms import StringField
+from wtforms import HiddenField
+from wtforms.validators import InputRequired, ValidationError
+from app import db
+from app.models import Artist, Album
+
+
+# Setup the Blueprint
+albums_bp = Blueprint(
+    "albums_bp", __name__, template_folder="templates", static_folder="static"
+)
+
+
+def does_album_exist(form, field):
+    album = (
+        db.session.query(Album)
+        .join(Artist)
+        .filter(Artist.name == form.artist.data)
+        .filter(Album.title == field.data)
+        .one_or_none()
+    )
+
+    if album is not None:
+        raise ValidationError("Album already exists", field.data)
+
+
+class CreateAlbumForm(FlaskForm):
+    artist = HiddenField("artist")
+    title = StringField(
+        label="Albums's Name", validators=[InputRequired(), does_album_exist]
+    )
+
+
+@albums_bp.route("/albums", methods=["GET", "POST"])
+@albums_bp.route("/albums/<int:artist_id>", methods=["GET", "POST"])
+def albums(artist_id=None):
+    form = CreateAlbumForm()
+
+    # Get the artist
+    artist = (
+        db.session.query(Artist)
+        .filter(Artist.artist_id == artist_id)
+        .one_or_none()
+    )
+
+    form.artist.data = artist.name
```

Comment:
If you click on _Albums_ in the top menu, no particular artist is chosen, so `artist` will be `None` after the query and `artist.name` will raise an `AttributeError`: _`'NoneType' object has no attribute 'name'`_.

I would probably have expected clicking on _Albums_ to show me a list of all albums in the database. If that is not trivial to fix/implement, one option could be to simply remove _Albums_ from the top menu. Accessing albums through the artist pages seem to work fine.